packages/browser-utils/src/getNativeImplementation.ts

```diff
@@ -0,0 +1,131 @@
+import { logger } from '@sentry/utils';
+import { DEBUG_BUILD } from './debug-build';
+import { WINDOW } from './types';
+
+/**
+ * We generally want to use window.fetch / window.setTimeout.
+ * However, in some cases this may be wrapped (e.g. by Zone.js for Angular),
+ * so we try to get an unpatched version of this from a sandboxed iframe.
+ */
+
+interface CacheableImplementations {
+  setTimeout: typeof WINDOW.setTimeout;
+  fetch: typeof WINDOW.fetch;
+}
+
+const cachedImplementations: Partial<CacheableImplementations> = {};
+
+/**
+ * isNative checks if the given function is a native implementation
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+function isNative(func: Function): boolean {
+  return func && /^function\s+\w+\(\)\s+\{\s+\[native code\]\s+\}$/.test(func.toString());
+}
```

Comment:
I added this, which is a generic version of this: https://github.com/getsentry/sentry-javascript/blob/develop/packages/browser/src/transports/utils.ts#L56-L58

Without this `tracing/metrics/web-vitals-fid` was failing consistently. Is the sandboxed version just too slow for our browser tests? cc @mydea

Reply:
hmm this looks reasonable. TBH the web-vitals tests are a bit of a black box to me, they appear/are brittle overall 😬 if that makes it pass, all good from my POV!